tests/test_waku_netconfig.nim

```diff
@@ -0,0 +1,297 @@
+    bindPort = conf.tcpPort,
+    dns4DomainName = some(dns4DomainName),
+    extPort = some(extPort)
+  )
+      
+  require:
+    netConfigRes.isOk()
+
+  let netConfig = netConfigRes.get()
+
+  check:
+    netConfig.announcedAddresses.len == 1  # Only DNS address should be present
+    netConfig.announcedAddresses[0] == dns4TcpEndPoint(dns4DomainName, extPort)
+
+asyncTest "AnnouncedAddresses includes extMultiAddrs when provided":
+
+    let 
+      conf = defaultTestWakuNodeConf()
+      extIp = ValidIpAddress.init("1.2.3.4")
+      extPort = Port(1234)
+      extMultiAddrs = @[ip4TcpEndPoint(extIp, extPort)]
+        
+    let netConfigRes = NetConfig.init(
+      bindIp = conf.listenAddress,
+      bindPort = conf.tcpPort,
+      extMultiAddrs = extMultiAddrs
+    )
+     
+    require:
+      netConfigRes.isOk()
+
+    let netConfig = netConfigRes.get()
+
+    check:
+      netConfig.announcedAddresses.len == 2  # Bind address + extAddress
+      netConfig.announcedAddresses[1] == extMultiAddrs[0]
+
+
+asyncTest "AnnouncedAddresses includes WebSocket addresses when enabled":
+
+    var 
+      conf = defaultTestWakuNodeConf()
+      wssEnabled = false
+        
+    var netConfigRes = NetConfig.init(
+      bindIp = conf.listenAddress,
+      bindPort = conf.tcpPort,
+      wsEnabled = true,
+      wssEnabled = wssEnabled
+    )
+     
+    require:
+      netConfigRes.isOk()
+
+    var netConfig = netConfigRes.get()
+
+    check:
+      netConfig.announcedAddresses.len == 2  # Bind address + wsHostAddress
+      netConfig.announcedAddresses[1] == (ip4TcpEndPoint(conf.listenAddress, 
+        conf.websocketPort) & wsFlag(wssEnabled))
+    
+    ## Now try the same for the case of wssEnabled = true
+        
+    wssEnabled = true
+    
+    netConfigRes = NetConfig.init(
+      bindIp = conf.listenAddress,
+      bindPort = conf.tcpPort,
+      wsEnabled = true,
+      wssEnabled = wssEnabled
+    )
+
+    require:
+      netConfigRes.isOk()
+
+    netConfig = netConfigRes.get()
+
+    check:
+      netConfig.announcedAddresses.len == 2  # Bind address + wsHostAddress
+      netConfig.announcedAddresses[1] == (ip4TcpEndPoint(conf.listenAddress, 
+        conf.websocketPort) & wsFlag(wssEnabled))
+    
+asyncTest "Announced WebSocket address contains external IP if provided":
+
+  let 
+    conf = defaultTestWakuNodeConf()
+    extIp = ValidIpAddress.init("1.2.3.4")
+    extPort = Port(1234)
+    wssEnabled = false
+        
+  let netConfigRes = NetConfig.init(
+    bindIp = conf.listenAddress,
+    bindPort = conf.tcpPort,
+    extIp = some(extIp),
+    extPort = some(extPort),
+    wsEnabled = true,
+    wssEnabled = wssEnabled
+  )
+     
+  require:
+    netConfigRes.isOk()
+
+  let netConfig = netConfigRes.get()
+
+  check:
+    netConfig.announcedAddresses.len == 2  # External address + wsHostAddress
+    netConfig.announcedAddresses[1] == (ip4TcpEndPoint(extIp, 
+      conf.websocketPort) & wsFlag(wssEnabled))
```

Comment:
Note that in order to configure WS using an `extIp`, the published port isn't `extPort` but `conf.websocketPort`. For websockets we bind to and announce the same port number.